dashboard/src/chain/errors.js

```diff
@@ -1,10 +1,12 @@
 const lib = {
   create: function(type, message, props = {}) {
     let err = new Error(message)
-    err = Object.assign(err, props, {
+    err = {
+      ...err,
+      ...props.data,
```

Comment:
Just by eyeballing it, I don't know if this will work.

- I think we want to do `Object.assign` here so it preserves the `Error` constructor and associated prototype.
- `props` looks like it's being used by the caller to inject arbitrary data into the error object. In other words, it is not the body of the error response. I think we have to update the request method itself to do that: https://github.com/chain/chain/blob/main/dashboard/src/chain/client.js#L89